bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerService.kt

```diff
@@ -124,12 +124,21 @@ class BusinessPartnerService(
         val partnerToUpsert = existingPartner ?: BusinessPartnerDb.createEmpty(upsertData.sharingState, upsertData.stage)
 
         val hasChanges = compareUtil.hasChanges(upsertData, partnerToUpsert)
+        val shouldUpdate = if(upsertData.currentness == null) {
```

Comment:
Refactor the upsertFromEntity function: Improve readability and efficiency by avoiding redundant null checks and using more idiomatic Kotlin. We can have something like below,
```
val hasChanges = compareUtil.hasChanges(upsertData, partnerToUpsert)
val shouldUpdate = when {
       upsertData.currentness == null -> true
       existingPartner?.currentness == null -> true
       else -> upsertData.currentness.isAfter(existingPartner.currentness)
 }

 if (hasChanges && shouldUpdate) {
       changelogRepository.save(ChangelogEntryDb(sharingState.externalId, sharingState.tenantBpnl, changeType, stage))
       copyUtil.copyValues(upsertData, partnerToUpsert)
       businessPartnerRepository.save(partnerToUpsert)
}

```